src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SqlConnection.cs

```diff
@@ -71,30 +71,34 @@ private enum CultureCheckState : uint
         // using SqlConnection.Open() method.
         internal bool _applyTransientFaultHandling = false;
 
+        // status of invariant culture environment check
+        private static CultureCheckState _cultureCheckState;
+
         // System column encryption key store providers are added by default
         private static readonly Dictionary<string, SqlColumnEncryptionKeyStoreProvider> s_systemColumnEncryptionKeyStoreProviders
-            = new Dictionary<string, SqlColumnEncryptionKeyStoreProvider>(capacity: 1, comparer: StringComparer.OrdinalIgnoreCase)
+            = new Dictionary<string, SqlColumnEncryptionKeyStoreProvider>(capacity: 3, comparer: StringComparer.OrdinalIgnoreCase)
                 {
                     {SqlColumnEncryptionCertificateStoreProvider.ProviderName, new SqlColumnEncryptionCertificateStoreProvider()},
                     {SqlColumnEncryptionCngProvider.ProviderName, new SqlColumnEncryptionCngProvider()},
                     {SqlColumnEncryptionCspProvider.ProviderName, new SqlColumnEncryptionCspProvider()}
                 };
 
+        /// Instance-level list of custom key store providers. It can be set more than once by the user.
+        private IReadOnlyDictionary<string, SqlColumnEncryptionKeyStoreProvider> _customColumnEncryptionKeyStoreProviders;
+
+        internal bool HasColumnEncryptionKeyStoreProvidersRegistered =>
+            _customColumnEncryptionKeyStoreProviders != null && _customColumnEncryptionKeyStoreProviders.Count > 0;
+
         // Lock to control setting of s_globalCustomColumnEncryptionKeyStoreProviders
         private static readonly object s_globalCustomColumnEncryptionKeyProvidersLock = new object();
-        // status of invariant culture environment check
-        private static CultureCheckState _cultureCheckState;
 
         /// <summary>
         /// Global custom provider list should be provided by the user. We shallow copy the user supplied dictionary into a ReadOnlyDictionary.
         /// Global custom provider list can only supplied once per application.
         /// </summary>
         private static IReadOnlyDictionary<string, SqlColumnEncryptionKeyStoreProvider> s_globalCustomColumnEncryptionKeyStoreProviders;
 
-        /// <summary>
-        /// Per-connection custom providers. It can be provided by the user and can be set more than once. 
-        /// </summary> 
-        private IReadOnlyDictionary<string, SqlColumnEncryptionKeyStoreProvider> _customColumnEncryptionKeyStoreProviders;
+        private static string _akvProviderName = "AZURE_KEY_VAULT";
```

Comment:
NIT: just to keep naming convention same as other:
```suggestion
        private static string s_akvProviderName = "AZURE_KEY_VAULT";
```